scripts/rollup/build.js

```diff
@@ -1,47 +1,97 @@
-  const createBundle = require('./bundle');
+    return acc;
+  }, {});
 
-  const rollup = createRollup(options);
-  const bundle = createBundle(options);
+// Self calling function to allow async/await for readability
+(async () => {
 
-  rollup
-    .then(bundle)
-    .then(() => {
-      console.log(`${pkgJSON.name} in ${options.format} is DONE`);
-    })
-    .catch((error) => {
+  // Create dist folder
+  await mkdirAsync(join(cwd, 'dist')).catch(err => {
+    if (err.code !== 'EEXIST') {
+      throw Error(err);
+    }
+  });
+  
+  // Get info from package.json
+  const {
+    version,
+    rollup: rollupConfig = {},
+    dependencies = {}, devDependencies = {}, peerDependencies = {}
+  } = pkgJSON;
+  
+  // Figure out from package.json what dependencies to bundle
+  function exclusionFilter(name) {
+    return !(rollupConfig.bundledDependencies || []).includes(name);
+  }
+  const deps = Object.assign({}, devDependencies, peerDependencies, dependencies);
+  const external = Object.keys(deps)
+    .filter(exclusionFilter)
+    .filter(function (elem, index, self) {
+      return index === self.indexOf(elem);
+    });
+  
+  // The import stream is NodeJS specific and should alwats be marked as external
+  // Used in inferno-server
+  external.push('stream');
+  
+  const defaultOptions = {
+    name: 'index',
+    replace: true,
+    version: pkgJSON.version
+  }
+  
+  const targets = [
+    //esmDev --name=index --ext=.dev.esm.js --env=development --format=es --minify=false
+    Object.assign({}, defaultOptions, { env: 'development', format: 'es', minify: false, ext: '.dev.esm.js'}),
+    //esmProd --name=index --ext=.esm.js --env=production --format=es --minify=false
+    Object.assign({}, defaultOptions, { env: 'production', format: 'es', minify: false, ext: '.esm.js'}),
+    //esNext --name=index --ext=.esnext.js --env=production --format=es --esnext=true --minify=false
+    Object.assign({}, defaultOptions, { env: 'production', format: 'es', esnext: true, minify: false, ext: '.esnext.js'}),
+    //cjsDev --env=development --format=cjs --replace=true --name=index.cjs --minify=false
+    Object.assign({}, defaultOptions, { env: 'development', format: 'cjs', minify: false, ext: '.cjs.js'}),
+    //cjsProd --env=production --format=cjs --replace=true --name=index.cjs --minify=true --ext=.min.js
+    Object.assign({}, defaultOptions, { env: 'production', format: 'cjs', minify: true, ext: '.cjs.min.js'}),
+    //umdDev --minify=false
+    Object.assign({}, defaultOptions, { env: 'development', format: 'umd', minify: false, name: pkgJSON.name, ext: '.js'}),
+    //umdProd --env=production --ext=.min.js
+    Object.assign({}, defaultOptions, { env: 'production', format: 'umd', minify: true, name: pkgJSON.name, ext: '.min.js'}),
+  ].filter((target) => {
+    if (args['target-ext']) {
+      return target.ext === args['target-ext'];
+    }
+    return true
+  })
+  
+  await targets.forEach(async (options) => {
```

Comment:
I wonder if the alias plugin is needed anymore? Also would it be possible to build all packages using rollup inputs rather than spawning new process for each package using lerna?

Reply:
Interesting! I didn't investigate the alias plugin due to lack of time. I ran into an issue that I know realise might have been related to that plugin but it solved itself. If we can remove it we should because it makes the build process harder to understand.

I am not sure that I understand what you mean by "using rollup inputs". I chose to let `lerna exec` take care of parallel execution because I wanted to simplify everything to allow a lower barrier of entry for further optimisation. If esbuild solves const enums I believe the esm and esmnext parts of the transforms could be done using esbuild without making the build process more complex, that would cut build time to 2.5mins.

Reply:
alias.js appears to be important to the build workflow. It allows packages to use transpiled output during bundling.